src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -5,19 +5,44 @@
  * максимального та мінімального елементів.
  */
 public class MatrixOperation {
+
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] diagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            diagonal[i] = matrix[i][i];
+        }
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int[matrix.length];
+        for (int i = matrix.length - 1; i >= 0; i--) {
+            counterDiagonal[matrix.length - i - 1] = matrix[matrix.length - i - 1][i];
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = matrix[0][0];
+        for (int[] ints : matrix) {
+            for (int anInt : ints) {
```

Comment:
let's rename `anInt` to `number` or `element` or smth else

Reply:
done